### DISCUSSION TITLE
No results returned from ldapsearch queries for new postgres based container install. What next?

### DISCUSSION THREAD

**Discussion by tetricky** [lldap/lldap#693] [Q&A] [answered]
I have an old lldap installation running (on old versions) using sqlite, in a container in a podman environment, working just fine.

I have a new server, set up for dual-stack, that I want to migrate over to, and I am setting up a new lldap environment, but this time using the new postgres option.

Using essentially the same setup, but with a .co.uk domain (I have also tried with just a .uk domain), and adding the appropriate `LLDAP_DATABASE_URL` I have set this up. The postgres database is populated, and I can check that users exist. I have set up a test user 'domainuser' and group 'site' so that I don't have just the admin user and groups.

From the host server, and other containers, that can reach the lldap server, the only response to ldapsearch queries I get is:

```
dn: ou=people,dc=domain,dc=co,dc=uk
objectClass: top
objectClass: organizationalUnit
```

I thought there might be some networking problems (although I verified connectivity from the various locations, and where the server wasn't available I got an error, rather than the empty response).

So I exec'd into the container, loaded openldap _clients ( `apk add openldap-clients` ) and performed a search from the lldap container itself. Same result. An example of my query is of the form:

```
ldapsearch -H ldap://lldap-domain:3890 -LLL -D "uid=admin,ou=people,dc=domain,dc=co,dc=uk" -w '<password>' -s "Base" -b "ou=people,dc=domain,dc=co,dc=uk" '(&(objectClass=person)(uid=domainuser))'
````

Same result, connection happens, no search results populated.

lldap-domain is the container name, and I have checked dns from within the container with ping (ipv4).

If I do a healthcheck in the container, all seems to check out okay.
```
/app # ./lldap healthcheck
Loading configuration from lldap_config.toml
2023-09-28T12:54:28.506994992+00:00  INFO     ｉ [info]: Starting healthchecks
2023-09-28T12:54:28.507447006+00:00  INFO     check_ldaps [ 16.7µs | 100.00% ] port: 6360
2023-09-28T12:54:28.507464103+00:00  INFO     ┕━ ｉ [info]: LDAPS not enabled
2023-09-28T12:54:28.507187575+00:00  INFO     check_ldap [ 1.17ms | 100.00% ] port: 3890
2023-09-28T12:54:28.542277197+00:00  INFO     ┕━ ｉ [info]: Success
2023-09-28T12:54:28.507557072+00:00  INFO     check_api [ 35.2ms | 100.00% ] port: 17170
2023-09-28T12:54:28.546293869+00:00  INFO     ┕━ ｉ [info]: Success
```

I'm stumped, and don't know where to go next to try to solve this.

I am far from an ldap expert, just trying to get a functional setup for a backend to authelia (again).

Any suggestions appreciated.

**Comment by nitnelave** [comment, marked as answer]
That is indeed the correct answer from the container (fixed from the _incorrect_ answer that the previous LLDAP versions would return):
You're asking about the `ou=people, ...` object, with an explicit scope of `Base`, meaning that you're only interested in the top-level object, not its descendants. If you change the scope to wholeSubtree (or "sub"), you'll get the list of all users.

Actually, I just realized that in the case of "sub" scope I'm also supposed to return the `ou=people` object.

See https://ldap.com/the-ldap-search-operation/

**Comment by tetricky** [comment]
Honestly. I don't know how you expect idiots like me to keep up, if you go around fixing things.

That was indeed a correct answer, to my incorrect question. Thank you very much.

If this now respects base/subtree, does this mean that lldap would now be suitable for holding user information for more than one domain? I was planning on a string of different lldap containers for different domains (hosted on the same server)...but it would be much better if I could just use one.

I've scanned the documentation and not seen anything that would point to this. It would be a lot more resource efficient for me.

**Comment by nitnelave** [comment]
No, there's no plan to have more than one domain.

What you can do:
- have all the users in the same LLDAP, have a group per domain
- run many LLDAP servers, they're really not that resource intensive (or is your experience otherwise?)

I also know that @Firstyear wrote an LDAP proxy, that might help if it supports several backends.

**Reply by Firstyear** [reply]
It does and it doesn't. It can only connect to one external ldap source, but nothing stops it serving data from multiple backends if they provider had them. So in the LLDAP case, it will only act as a query cache / firewall, rather than allowing "knitting" together multiple instances.

**Comment by tetricky** [comment]
That's fine, I'll go with different lldap servers. It's more the overhead of lots of different containers/databases, than the resource footprint of lldap itself. But the simplicity works for me, better than the minor inconvenience of multiple containers.

Your quick response very much appreciated.

**Reply by nitnelave** [reply]
Note that you can use the same postgres db, with a separate database for each server
